pkg/kubelet/prober/worker.go

```diff
@@ -316,6 +325,29 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		return true
 	}
 
+	if resultChanged {
+		var eventType string
+		var eventReason string
+		if result == results.Success {
+			eventType = v1.EventTypeNormal
+			eventReason = events.ContainerProbeResultSuccess
+		} else {
+			eventType = v1.EventTypeWarning
+			eventReason = events.ContainerProbeResultFailure
+		}
+		RecordContainerEvent(
+			w.recorder,
+			w.pod,
+			&w.container,
+			eventType,
+			eventReason,
+			"%s probe result changed to %s%s",
```

Comment:
message may be a bit weird for startup probe.

Reply:
It should be "Startup probe result changed to Success" or "Startup probe result changed to Failure: error information",  to me that feels like a useful message for users?